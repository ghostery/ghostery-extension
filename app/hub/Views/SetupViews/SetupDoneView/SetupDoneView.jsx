/**
 * Setup Done View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

/**
 * A Functional React component for rendering the Setup Done View
 * @return {JSX} JSX for rendering the Setup Done View of the Hub app
 * @memberof HubComponents
 */
const SetupDoneView = props => (
	<div className="row align-center">
		<div className="columns small-12 large-10">
			<div className="SetupDone">
				<div className="SetupDone__title_bar row align-middle">
					<div className="SetupDone__bar columns" />
					<div className="SetupDone__title columns shrink">
						{t('hub_setup_ready_heading')}
					</div>
					<div className="SetupDone__bar columns" />
				</div>

				<div className="row align-spaced">
					{props.features.map((feature) => {
						const iconClassNames = `SetupDone__feature_icon feature-${feature.id}`;

						return (
							<div key={`feature-${feature.id}`} className="SetupDone__feature columns flex-container flex-dir-column">
								<div className={iconClassNames} />
								<div className="SetupDone__feature_title flex-container align-center-middle">
									{feature.title}
								</div>
								<div className="SetupDone__bar" />
								<div className="flex-child-grow">{feature.description}</div>
								<NavLink className="SetupDone__feature_button button primary" to={feature.buttonHref}>
									{feature.buttonText}
								</NavLink>
							</div>
						);
					})}
				</div>

			</div>
		</div>
	</div>
);

// PropTypes ensure we pass required props of the correct type
SetupDoneView.propTypes = {
	features: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		description: PropTypes.string.isRequired,
		buttonText: PropTypes.string.isRequired,
		buttonHref: PropTypes.string.isRequired,
	})).isRequired,
};

export default SetupDoneView;
