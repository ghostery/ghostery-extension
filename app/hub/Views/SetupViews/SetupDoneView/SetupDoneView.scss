/**
 * Setup Done View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Setup Blocking View
.SetupDone__title_bar {
	margin-bottom: 25px;
}
.SetupDone__bar {
	background-color: #979797;
	height: 1px;
}
.SetupDone__title {
	color: #4a4a4a;
	font-size: 16px;
	line-height: 42px;
	padding: 0 25px;
	margin: 0;
}
.SetupDone__feature {
	background-color: #f0f3f4;
	color: #4a4a4a;
	padding: 27px 15px 17px;
	max-width: 177px;
	min-height: 273px;
	border-radius: 4px;
	text-align: center;
}
.SetupDone__feature_icon {
	height: 50px;
	width: 50px;
	margin: 0 auto;
	background-repeat: no-repeat;
	background-position: center center;
}
.SetupDone__feature_icon.feature-tutorial {
	background-image: buildFeatureTutorialIcon(#4a4a4a);
}
.SetupDone__feature_icon.feature-supporter {
	background-image: buildFeatureSupporterIcon(#4a4a4a);
}
.SetupDone__feature_icon.feature-products {
	background-image: buildFeatureProductsIcon(#4a4a4a);
}
.SetupDone__feature_title {
	margin-top: 10px;
	font-size: 14px;
	font-weight: 500;
	line-height: 22px;
	min-height: 45px;
}
.SetupDone__feature .SetupDone__bar {
	margin: 10px;
}
.SetupDone__feature_button {
	margin: 20px 10px 10px;
}
