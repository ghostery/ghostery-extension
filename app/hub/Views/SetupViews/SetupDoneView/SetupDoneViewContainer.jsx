/**
 * Setup Done View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SetupDoneView from './SetupDoneView';

/**
 * @class Implement the Setup Done View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class SetupDoneViewContainer extends Component {
	/**
	 * Lifecycle Event
	 */
	componentWillMount() {
		const title = t('hub_setup_page_title_done');
		window.document.title = title;

		const { index } = this.props;
		this.props.actions.setSetupNavigation({
			activeIndex: index,
			hrefPrev: `/setup/${index - 1}`,
			hrefNext: '/',
			hrefDone: false,
			textPrev: t('hub_setup_nav_previous'),
			textNext: t('hub_setup_nav_done'),
			textDone: false,
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Setup Done View of the Hub app
	 */
	render() {
		const features = [
			{
				id: 'tutorial',
				title: t('hub_setup_feature_tutorial_title'),
				description: t('hub_setup_feature_tutorial_description'),
				buttonText: t('hub_setup_feature_tutorial_button'),
				buttonHref: '/tutorial/1',
			},
			{
				id: 'supporter',
				title: t('hub_setup_feature_supporter_title'),
				description: t('hub_setup_feature_supporter_description'),
				buttonText: t('hub_setup_feature_supporter_button'),
				buttonHref: '/supporter',
			},
			{
				id: 'products',
				title: t('hub_setup_feature_products_title'),
				description: t('hub_setup_feature_products_description'),
				buttonText: t('hub_setup_feature_products_button'),
				buttonHref: '/products',
			},
		];
		return <SetupDoneView features={features} />;
	}
}

// PropTypes ensure we pass required props of the correct type
SetupDoneViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setSetupNavigation: PropTypes.func.isRequired,
	}).isRequired,
};

export default SetupDoneViewContainer;
