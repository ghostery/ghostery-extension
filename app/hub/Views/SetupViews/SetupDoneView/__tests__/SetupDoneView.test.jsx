/**
 * Setup Done View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import SetupDoneView from '../SetupDoneView';

describe('app/hub/Views/SetupViews/SetupDoneView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup human web view is rendered correctly', () => {
			const initialState = {
				features: [
					{
						id: 'feature-1',
						title: 'Feature 1 Title',
						description: 'feature 1 description',
						buttonText: 'feature 1 button text',
						buttonHref: '/feature/1',
					},
					{
						id: 'feature-2',
						title: 'Feature 2 Title',
						description: 'feature 2 description',
						buttonText: 'feature 2 button text',
						buttonHref: '/feature/2',
					},
					{
						id: 'feature-3',
						title: 'Feature 3 Title',
						description: 'feature 3 description',
						buttonText: 'feature 3 button text',
						buttonHref: '/feature/3',
					},
				],
			};

			const component = renderer.create(
				<MemoryRouter>
					<SetupDoneView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				features: [
					{
						id: 'feature-1',
						title: 'Feature 1 Title',
						description: 'feature 1 description',
						buttonText: 'feature 1 button text',
						buttonHref: '/feature/1',
					},
					{
						id: 'feature-2',
						title: 'Feature 2 Title',
						description: 'feature 2 description',
						buttonText: 'feature 2 button text',
						buttonHref: '/feature/2',
					},
					{
						id: 'feature-3',
						title: 'Feature 3 Title',
						description: 'feature 3 description',
						buttonText: 'feature 3 button text',
						buttonHref: '/feature/3',
					},
				],
			};

			const component = shallow(<SetupDoneView {...initialState} />);
			expect(component.find('.SetupDone').length).toBe(1);
		});
	});
});
