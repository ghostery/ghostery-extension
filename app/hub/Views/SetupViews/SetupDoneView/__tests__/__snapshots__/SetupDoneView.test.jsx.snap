// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupViews/SetupDoneView component Snapshot tests with react-test-renderer setup human web view is rendered correctly 1`] = `
<div
  className="row align-center"
>
  <div
    className="columns small-12 large-10"
  >
    <div
      className="SetupDone"
    >
      <div
        className="SetupDone__title_bar row align-middle"
      >
        <div
          className="SetupDone__bar columns"
        />
        <div
          className="SetupDone__title columns shrink"
        >
          hub_setup_ready_heading
        </div>
        <div
          className="SetupDone__bar columns"
        />
      </div>
      <div
        className="row align-spaced"
      >
        <div
          className="SetupDone__feature columns flex-container flex-dir-column"
        >
          <div
            className="SetupDone__feature_icon feature-feature-1"
          />
          <div
            className="SetupDone__feature_title flex-container align-center-middle"
          >
            Feature 1 Title
          </div>
          <div
            className="SetupDone__bar"
          />
          <div
            className="flex-child-grow"
          >
            feature 1 description
          </div>
          <a
            aria-current={null}
            className="SetupDone__feature_button button primary"
            href="/feature/1"
            onClick={[Function]}
            style={undefined}
          >
            feature 1 button text
          </a>
        </div>
        <div
          className="SetupDone__feature columns flex-container flex-dir-column"
        >
          <div
            className="SetupDone__feature_icon feature-feature-2"
          />
          <div
            className="SetupDone__feature_title flex-container align-center-middle"
          >
            Feature 2 Title
          </div>
          <div
            className="SetupDone__bar"
          />
          <div
            className="flex-child-grow"
          >
            feature 2 description
          </div>
          <a
            aria-current={null}
            className="SetupDone__feature_button button primary"
            href="/feature/2"
            onClick={[Function]}
            style={undefined}
          >
            feature 2 button text
          </a>
        </div>
        <div
          className="SetupDone__feature columns flex-container flex-dir-column"
        >
          <div
            className="SetupDone__feature_icon feature-feature-3"
          />
          <div
            className="SetupDone__feature_title flex-container align-center-middle"
          >
            Feature 3 Title
          </div>
          <div
            className="SetupDone__bar"
          />
          <div
            className="flex-child-grow"
          >
            feature 3 description
          </div>
          <a
            aria-current={null}
            className="SetupDone__feature_button button primary"
            href="/feature/3"
            onClick={[Function]}
            style={undefined}
          >
            feature 3 button text
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
